scripts/migrations/2025_05_convert_sto_to_json.pl

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/perl -w
+
+# This file is part of Product Opener.
+#
+# Product Opener
+# Copyright (C) 2011-2025 Association Open Food Facts
+# Contact: contact@openfoodfacts.org
+# Address: 21 rue des Iles, 94100 Saint-Maur des Fossés, France
+#
+# Product Opener is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use ProductOpener::PerlStandards;
+use utf8;
+
+use ProductOpener::Paths qw/%BASE_DIRS/;
+use ProductOpener::Store qw/object_iter retrieve_object store_object/;
+
+# This script converts all product sto files to json
+
+my ($checkpoint_file, $last_processed_path) = open_checkpoint('2025_05_convert_sto_to_json.tmp');
```

Comment:
NB: this file was accidentally committed to source control on the branch.  I wonder whether we could update `open_checkpoint` to place checkpoint files within `/mnt/podata/tmp`?

Reply:
It wasn't an accident, but just not something to be run just yet. I have noticed we have a lot of scripts that use checkpoints (and those that don't possibly should) so it might be a good idea to centralise this logic somewhere. Unfortunately not something I'll be able to get to before the Perl hack weekend

Reply:
Ok, got it.  I did wonder after commenting whether I might have been mistaken and that it could have been intentional for some reason.

From what I remember: the checkpoint _code_ is reasonably centralised already, but the deployment configuration seems variable (e.g. some of the scripts rely on being started from a particular working directory, which does not seem ideal).

Reply:
Ah: I mistakenly opened this thread on the script, instead of the `.tmp` file.  You did resolve my concerns about it though :)  Please consider this resolved.